src/ec2/explorer/ec2InstanceNode.ts

```diff
@@ -0,0 +1,44 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { getNameOfInstance } from '../../shared/clients/ec2Client'
+import { AWSResourceNode } from '../../shared/treeview/nodes/awsResourceNode'
+import { AWSTreeNodeBase } from '../../shared/treeview/nodes/awsTreeNodeBase'
+import { Ec2Instance } from '../../shared/clients/ec2Client'
+import globals from '../../shared/extensionGlobals'
+export class Ec2InstanceNode extends AWSTreeNodeBase implements AWSResourceNode {
+    public constructor(
+        public override readonly regionCode: string,
+        private instance: Ec2Instance,
+        public override readonly contextValue: string
+    ) {
+        super('')
+        this.update(instance)
+    }
+
+    public update(newInstance: Ec2Instance) {
+        this.setInstance(newInstance)
+        this.label = this.name
+        this.tooltip = this.InstanceId
+    }
+
+    public setInstance(newInstance: Ec2Instance) {
+        this.instance = newInstance
+    }
+
+    public get name(): string {
+        return getNameOfInstance(this.instance) ?? 'Unnamed instance'
+    }
+
+    public get InstanceId(): string {
+        return this.instance.InstanceId!
+    }
+
+    public get arn(): string {
+        return `arn:aws:ec2:${this.regionCode}:${globals.awsContext.getCredentialAccountId()}:instance/${
```

Comment:
does `Ec2Instance` or some other SDK function not have a arn() method?

Reply:
I don't believe so, or at least I couldn't find it. https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-instance.html.

Reply:
ok, as noted in that post we should avoid hardcoding "aws". 

- example: https://github.com/aws/aws-toolkit-vscode/blob/d66f115ee4e1fed8d87fa6a59bd83a1c5925ac05/src/apigateway/explorer/apiNodes.ts#L45
- the partition is passed from the explorer tree root: https://github.com/aws/aws-toolkit-vscode/blob/d66f115ee4e1fed8d87fa6a59bd83a1c5925ac05/src/awsexplorer/regionNode.ts#L33
- it might be more robust if we use the sdk's "builder" https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/functions/_aws_sdk_util_arn_parser.build.html